test/unit/object/test_chunked_upload.py

```diff
@@ -0,0 +1,132 @@
+        'size': 2,
+        'sha1': second_sha1,
+    }
+    part_three = {
+        'part_id': '6F2D3486',
+        'offset': 4,
+        'size': 2,
+        'sha1': third_sha1,
+    }
+    part_four = {
+        'part_id': '4DBC872D',
+        'offset': 6,
+        'size': 2,
+        'sha1': fourth_sha1,
+    }
+    parts = [part_one, part_two, part_three, part_four]
+    expected_data = {
+        'parts': parts,
+    }
+    expected_headers = {
+        'Content-Type': 'application/json',
+        'Digest': 'SHA=QlrxKgdDUCsyLpOgFbz4aOMk1Wo=',
+    }
+
+    expected_headers_first_upload = {
+        'Content-Type': 'application/octet-stream',
+        'Digest': 'SHA={}'.format(first_sha1),
+        'Content-Range': 'bytes 0-1/8',
+    }
+    expected_headers_second_upload = {
+        'Content-Type': 'application/octet-stream',
+        'Digest': 'SHA={}'.format(second_sha1),
+        'Content-Range': 'bytes 2-3/8',
+    }
+    expected_headers_third_upload = {
+        'Content-Type': 'application/octet-stream',
+        'Digest': 'SHA={}'.format(third_sha1),
+        'Content-Range': 'bytes 4-5/8',
+    }
+    expected_headers_fourth_upload = {
+        'Content-Type': 'application/octet-stream',
+        'Digest': 'SHA={}'.format(fourth_sha1),
+        'Content-Range': 'bytes 6-7/8',
+    }
+
+    first_response_mock = Mock()
+    second_response_mock = Mock()
+    third_response_mock = Mock()
+    fourth_response_mock = Mock()
+
+    first_response_mock.json.return_value = {
+        'part': part_one
+    }
+    second_response_mock.json.return_value = {
+        'part': part_two
+    }
+    third_response_mock.json.return_value = {
+        'part': part_three
+    }
+    fourth_response_mock.json.return_value = {
+        'part': part_four
+    }
+    mock_box_session.put.side_effect = [first_response_mock, second_response_mock,
+                                        third_response_mock, fourth_response_mock]
+    mock_box_session.post.return_value.json.return_value = {
+        'entries': [
+            {
+                'type': 'file',
+                'id': '12345',
+                'description': 'This is a test description',
+            }
+        ]
+    }
+    chunked_uploader = ChunkedUpload(test_upload_session, stream, file_size)
+    uploaded_file = chunked_uploader.start()
+    calls = [call(expected_put_url, data=b'ab', headers=expected_headers_first_upload),
+             call(expected_put_url, data=b'cd', headers=expected_headers_second_upload),
+             call(expected_put_url, data=b'ef', headers=expected_headers_third_upload),
+             call(expected_put_url, data=b'gh', headers=expected_headers_fourth_upload), ]
+    mock_box_session.put.assert_has_calls(calls, any_order=False)
+    mock_box_session.post.assert_called_once_with(expected_post_url, data=json.dumps(expected_data),
+                                                  headers=expected_headers)
+    assert uploaded_file.type == 'file'
```

Comment:
It would be good to also assert that `uploaded_file` is actually a `File` and that its `_session` is `mock_box_session`